Update JFrog GitHub OIDC setup docs (#37596)
github/docs: content/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-jfrog.md

## Updating your {% data variables.product.prodname_actions %} workflow

> Once you establish a trust relationship between {% data variables.product.prodname_actions %} and the JFrog platform, you can update your {% data variables.product.prodname_actions %} workflow file.

Reviewer: Is this information not accurate any more? It seems to add a nice bridge between the H2 and the H3

In your {% data variables.product.prodname_actions %} workflow file, ensure you are using the provider name and audience you configured in the JFrog Platform.

The following example uses the placeholders `YOUR_PROVIDER_NAME` and `YOUR_AUDIENCE`.

```yaml
permissions:
  id-token: write
  contents: read

jobs:
  build:
    runs-on: ubuntu-latest
    steps:

>       - name: Setup JFrog CLI with OIDC

Reviewer: ```suggestion
      - name: Set up JFrog CLI with OIDC
```

        id: setup-jfrog-cli
        uses: jfrog/setup-jfrog-cli@29fa5190a4123350e81e2a2e8d803b2a27fed15e
        with:
          JF_URL: ${{ env.JF_URL }}
          oidc-provider-name: 'YOUR_PROVIDER_NAME' 
          oidc-audience: 'YOUR_AUDIENCE' # This is optional

      - name: Upload artifact
        run: jf rt upload "dist/*.zip" my-repo/

```

> [!TIP]
> When OIDC authentication is used, the `setup-jfrog-cli` action automatically provides `oidc-user` and `oidc-token` as step outputs.
> These can be used for other integrations that require authentication with JFrog.

> > To reference these outputs, ensure the step has an explicit `id` defined (e.g., `id: setup-jfrog-cli`).

Reviewer: ```suggestion
> To reference these outputs, ensure the step has an explicit `id` defined (for example `id: setup-jfrog-cli`).
```

> ### Example: Using OIDC Credentials in Other Steps

Reviewer: ```suggestion
### Using OIDC Credentials in Other Steps
```

> ### Example: Using OIDC Credentials in Other Steps

Reviewer: Is there a bit of extra context you could add here, just to set the scene for the example?

## Updating your {% data variables.product.prodname_actions %} workflow

### Example: Authenticating with JFrog using OIDC

In your {% data variables.product.prodname_actions %} workflow file, ensure you are using the provider name and audience you configured in the JFrog Platform.

```yaml
permissions:
  id-token: write
  contents: read

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Setup JFrog CLI with OIDC
        uses: jfrog/setup-jfrog-cli@v4
        with:
          oidc-provider-name: 'YOUR_PROVIDER_NAME'
          oidc-audience: 'YOUR_AUDIENCE'

      - name: Upload artifact
        run: jf rt upload "dist/*.zip" my-repo/
```

## Security Best Practices

- Always use `permissions: id-token: write` in workflows that authenticate with JFrog.
- Limit trust using specific claims like `repository`, `ref`, or `environment`.

> - Configure identity mappings in JFrog to restrict which workflows are allowed to authenticate.

Reviewer 2: Also, maybe rephrase as: "Configure identity mappings in JFrog to restrict authentication to specific workflows."

> - Configure identity mappings in JFrog to restrict which workflows are allowed to authenticate.

Author: Great! rephrased and added the link the further reading

## Updating your {% data variables.product.prodname_actions %} workflow

### Example: Authenticating with JFrog using OIDC

In your {% data variables.product.prodname_actions %} workflow file, ensure you are using the provider name and audience you configured in the JFrog Platform.

> The following example uses the placeholders YOUR_PROVIDER_NAME and YOUR_AUDIENCE.

Reviewer 3: Code blocks (`) missing around the placeholders. Intentional?

```yaml
permissions:
  id-token: write
  contents: read

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Setup JFrog CLI with OIDC
        id: setup-jfrog-cli

>         uses: jfrog/setup-jfrog-cli@v4

Reviewer 3: Rather than `@v4`, could we please have the example pin to the equivalent full SHA? That matches our other provider docs (e.g. [AWS](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#requesting-the-access-token), [Hashicorp](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-hashicorp-vault#requesting-the-access-token), [PyPI](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-pypi#updating-your-github-actions-workflow)) and is more consistent with our guidance at https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions.

## Updating your {% data variables.product.prodname_actions %} workflow

### Example: Authenticating with JFrog using OIDC

In your {% data variables.product.prodname_actions %} workflow file, ensure you are using the provider name and audience you configured in the JFrog Platform.

The following example uses the placeholders YOUR_PROVIDER_NAME and YOUR_AUDIENCE.

```yaml
permissions:
  id-token: write
  contents: read

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Setup JFrog CLI with OIDC
        id: setup-jfrog-cli
        uses: jfrog/setup-jfrog-cli@v4
        with:
          JF_URL: ${{ env.JF_URL }}
          oidc-provider-name: 'YOUR_PROVIDER_NAME' 
          oidc-audience: 'YOUR_AUDIENCE'

      - name: Upload artifact
        run: jf rt upload "dist/*.zip" my-repo/

```

> [!TIP]
> When OIDC authentication is used, the `setup-jfrog-cli` action automatically provides `oidc-user` and `oidc-token` as step outputs.

> > These can be used for other integrations that require authentication with JFrog.

Reviewer 2: Maybe clarify "These outputs"?

> To reference these outputs, ensure the step has an explicit `id` defined (e.g., `id: setup-jfrog-cli`).

### Example: Using OIDC Credentials in Other Steps
```yaml
      - name: Login to Artifactory Docker Registry
        uses: docker/login-action@v3
        with:
          registry: ${{ env.JF_URL }}
          username: ${{ steps.setup-jfrog-cli.outputs.oidc-user }}
          password: ${{ steps.setup-jfrog-cli.outputs.oidc-token }}
```
## Security Best Practices

-  Always set `permissions: id-token: write` in any workflow using OIDC.
-  Restrict trust using claims such as `repository`, `ref`, or `environment`.
-  Use identity mappings to scope access at a fine-grained level inside the JFrog Platform.

## Further Reading

> 

Reviewer 2: Possibly add the documentation for dynamic mapping, or token creation

Reviewer 4: All over right

In your {% data variables.product.prodname_actions %} workflow file, ensure you are using the provider name and audience you configured in the JFrog Platform.

> The following example uses the placeholder `YOUR_PROVIDER_NAME` & `YOUR_AUDIENCE`.

Reviewer 2: Please correct to: "The following example uses the placeholders`YOUR_PROVIDER_NAME` and `YOUR_AUDIENCE`"
